Tests/subset/subset_test.py

```diff
@@ -1352,5 +1352,55 @@ def test_subset_svg_missing_lxml(ttf_path):
         subset.main([str(ttf_path), "--gids=0,1"])
 
 
+def test_subset_COLR_glyph_closure(tmp_path):
+    # https://github.com/fonttools/fonttools/issues/2461
+    font = TTFont()
+    ttx = pathlib.Path(__file__).parent / "data" / "BungeeColor-Regular.ttx"
+    font.importXML(ttx)
+
+
+    color_layers = font["COLR"].ColorLayers
+    assert ".notdef" in color_layers
+    assert "Agrave" in color_layers
+    assert "grave" in color_layers
+
+    font_path = tmp_path / "BungeeColor-Regular.ttf"
+    subset_path = font_path.with_suffix(".subset.ttf)")
+    font.save(font_path)
+
+    subset.main(
+        [
+            str(font_path),
+            "--glyph-names",
+            f"--output-file={subset_path}",
+            "--glyphs=Agrave",
+        ]
+    )
+    subset_font = TTFont(subset_path)
+
+    glyph_order = subset_font.getGlyphOrder()
+
+    assert glyph_order == [
+        ".notdef",  # '.notdef' is always included automatically
+        "A",
+        "grave",  # included because glyf component of Agrave
+        "Agrave",
+        ".notdef.alt001",
+        ".notdef.alt002",
+        "A.alt002",
+        "Agrave.alt001",
+        "Agrave.alt002",
+        "grave.alt002",
+    ]
```

Comment:
Seems like more glyphs than you should need to minimally demonstrate the problem? - IIUC two glyf entries, one a component of the other, and two COLR glyphs, each using one of the glyf entries, would suffice? - subset down to only the glyf that has the other as a component and we should see the COLR glyph corresponding to the component get dropped.

IMO it might clearer if glyphs were renamed as well, e.g. glyf_with_colr1, glyph_with_colr2, glyf_with_colr1.component1, ... or w/e, not tied to those names, just hoping clearer names might be possible.

Reply:
yes, maybe it could be more minimal to aid clarity, but I prefer to use an existing font instead of making one from scratch or fiddle with it. It's good enough.